helix-term/src/commands.rs

```diff
@@ -4578,8 +4578,12 @@ fn shell_impl(
     }
     let output = process.wait_with_output()?;
 
-    if !output.stderr.is_empty() {
-        log::error!("Shell error: {}", String::from_utf8_lossy(&output.stderr));
+    if !output.status.success() {
+        if !output.stderr.is_empty() {
+            let err = String::from_utf8_lossy(&output.stderr);
+            bail!("Shell error: {}", err.to_string());
+        }
+        bail!("Shell command failed");
     }
 
```

Comment:
With this change we loose the stderr going to `log::error!` if `output.status.success() == true`. We should always send stderr to the log:

- if the command succeeded, the stderr would otherwise be discarded
- if the command failed, the stderr may be longer than the editor's status element and the user may want to see the full stderr in the log

Reply:
>if the command succeeded, the stderr would otherwise be discarded

If the command succeeded, then is it necessary? If there was no error code, then it isn't necessarily an error. Some programs simply use it for international output, e.g. `xargs -t` will print the command it's running to stderr. `ssh -v` prints debug info to stderr. Despite its name, stderr is used for a lot more things than error messages. Maybe this could be useful as a `log::debug!` though.

> if the command failed, the stderr may be longer than the editor's status element and the user may want to see the full stderr in the log

Yeah it might be useful to add a `log::error!` at the [call site](https://github.com/helix-editor/helix/blob/4ce5a94552d887b9b14000dfdd1abe76a6d5f315/helix-term/src/commands.rs#L4610)